### PR TITLE
make --track mandatory

### LINKED ISSUE #1176: Require a track when running benchmarks
So far Rally has implicitly chosen the `geonames` track when `--track` was absent. A default value here is confusing because Rally just executes a seemingly random workload. There is also no particularly good choice for a default benchmark so we have deprecated this behavior in Rally 2.1.0 with #1175. In the next minor release we should remove this default and require users to explicitly specify a track when running benchmarks.

### PR DESCRIPTION
**Pull request by DJRickyB**
closes #1176

### CODE REVIEW

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks! I left one suggestion how we could make the error message look a bit more like an argparse error. I also checked the build failure and I think that this was a transient issue (`test_local_installation` has failed in the Python 3.9 build) so I'm going to retrigger CI.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please

File: esrally/rally.py [hunk 1]
```diff
@@ -710,15 +710,9 @@ def configure_track_params(arg_parser, args, cfg, command_requires_track=True):
...
-            # set the default programmatically because we need to determine whether the user has provided a value
-            if args.track:
-                chosen_track = args.track
-            else:
-                chosen_track = "geonames"
-                console.warn(f"Starting Rally without --track is deprecated. Add --track={chosen_track} to your parameters.")
-
-            cfg.add(config.Scope.applicationOverride, "track", "track.name", chosen_track)
+            if not args.track:
+                raise exceptions.SystemSetupError("Starting Rally without specifying --track is not supported.")
```
[the remark this answers is not in the document]
Comment (DJRickyB) [reply esrally/rally.py]: suggestion applied

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

**Comment by danielmitterdorfer** [comment]
@DJRickyB one more ask: Can you please ensure that your commit includes a body text that explains the motivation for this change? I think we should strive to include a description in all but the most trivial cases (ideally already when we raise the PR but at latest when we merge it) so it is easier to understand the project history.

**Pull request merged**